src/consensus/encode.rs

```diff
@@ -189,7 +189,7 @@ pub fn serialize<T: Encodable + ?Sized>(data: &T) -> Vec<u8> {
 
 /// Encode an object into a hex-encoded string
 pub fn serialize_hex<T: Encodable + ?Sized>(data: &T) -> String {
-    hex_encode(serialize(data))
+    serialize(data)[..].to_hex()
```

Comment:
This is probably gonna be a bit slower, but dunno if we really care tooo much. Everything else is just fmt::Display, which I think is fine.

Reply:
Hmm, I could have sworn we had something like a `HexWriter` in bitcoin_hashes that would let us serialize directly into a string.

Reply:
We do not appear to, no, sadly.

Reply:
Lol, I was confused with this issue that I opened but never addressed https://github.com/rust-bitcoin/bitcoin_hashes/issues/54